mock_service/lib/behavior.py

```diff
@@ -77,17 +88,18 @@ def execute(self, delta_time: float, animators):
             matched_datapoint = self._mocked_datapoints[path]
             for behavior in behaviors:
                 execution_context = ExecutionContext(
-                    path, self._pending_event_list, self._mocked_datapoints, delta_time
+                    path, self._pending_event_list, delta_time
                 )
-                trigger_result = behavior.check_trigger(execution_context)
-                if trigger_result.is_active() and behavior.is_condition_fulfilled(
-                    execution_context
-                ):
-                    log.info(f"Running behavior for {path}")
-                    behavior.execute(
-                        ActionContext(
-                            trigger_result, execution_context, matched_datapoint
-                        ),
-                        animators,
-                    )
-                    break
+                if behavior.is_condition_fulfilled(
```

Comment:
Is there any specific reason (i.e. a bug) that lead to the swapping of the condition check and trigger activation check?

Reply:
You are right, all conditions which listen to the same event in the same iteration should get triggered. We should raise a bug issue for this. For now I understand why this change is necessary.

Reply:
#91